File: docs/benchmarks/suite/fixtures/python-service/app/repositories/session_repository.py
from dataclasses import dataclass


@dataclass(frozen=True)
class SessionRecord:
    session_id: str
    user_id: str
    device_id: str


class SessionRepository:
    def create_login_session(self, user_id: str, device_id: str) -> SessionRecord:
        return SessionRecord(
            session_id=f"{user_id}:{device_id}",
            user_id=user_id,
            device_id=device_id,
        )

File: docs/benchmarks/suite/fixtures/go-service/internal/repository/session_repository.go
package repository

type Session struct {
	ID       string `json:"session_id"`
	UserID   string `json:"user_id"`
	DeviceID string `json:"device_id"`
}

type SessionRepository struct{}

func NewSessionRepository() *SessionRepository {
	return &SessionRepository{}
}

func (r *SessionRepository) CreateLoginSession(userID string, deviceID string) Session {
	return Session{
		ID:       userID + ":" + deviceID,
		UserID:   userID,
		DeviceID: deviceID,
	}
}
